python/lsst/ts/epm/mib_tree.py

```diff
@@ -0,0 +1,185 @@
+# This file is part of ts_epm.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the Vera Rubin Observatory
+# Project (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["mib_tree"]
+
+import logging
+import pathlib
+import re
+
+from .utils import MibTreeElement, MibTreeElementType
+
+OBJECT_IDENTIFIER = r"^(\w+) +OBJECT IDENTIFIER +::= \{ ?(\w+) (\d+) ?\}$"
+OBJECT_TYPE = r"^(\w+) OBJECT-TYPE$"
+MIB_OID = r"^::= ?{ ?(\w+) +(\d+) ?}$"
+DESCRIPTION = "DESCRIPTION"
+
+DATA_DIR = pathlib.Path(__file__).parent / "data"
+
+_line_num = 0
+
+log = logging.getLogger("MIB")
```

Comment:
Having to define a logger here at the module levels suggests that there's something wrong here. 

I really don't like the way you did this here. Having this much code run when the user is importing the module is really not a good idea. I would suggest you make a `MibTree` class or a `make_mib_tree` method that returns the `mib_tree` dictionary, and let the user control when this executes.  I think a class would make sense as you could add a logger etc.  You could even make your class work like a dictionary by implementing `__getitem__`.

I think it would be much more ergonomic if it worked like:

```
mib_tree = MibTree()  # all the code required to construct the class runs here.

# mib tree elements can be accessed as a dictionary by implementing  __getitem__
mib_tree["snmp"]
```

Reply:
I reworked the code so this is all in a class now.